code-orange/VisitedLocationPanel.swift

```diff
@@ -0,0 +1,233 @@
+//
+//  VisitedLocationPanel.swift
+//  code-orange
+//
+//  Created by Renen Elal on 16/03/2020.
+//  Copyright © 2020 Renen Elal. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+protocol VisitedLocationsPanelDelegate: class {
+  func visitedLocationsPanelCallEmergency()
+  func visitedLocationsPanelOpenHealthAdministrationWebsite()
+  func visitedLocationsDidSelectLocation(_ location: RecordedLocation)
+}
+
+class VisitedLocationsPanel: UIView {
+  
+  public weak var delegate: VisitedLocationsPanelDelegate?
+  private var currentLocationIndex = 0
+  private var locations: [RecordedLocation]
+
+  init?(locations: [RecordedLocation]) {
+    guard !locations.isEmpty else { return nil }
+    self.locations = locations
+    super.init(frame: .zero)
+    commonInit()
+  }
+  
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+  
+  private func commonInit() {
+    addSubview(mainStackView)
+    mainStackView.fillSuperview()
+    reloadData()
+  }
+  
+  private lazy var mainStackView: UIStackView = {
+    let stackView = UIStackView()
+    stackView.translatesAutoresizingMaskIntoConstraints = false
+    stackView.axis = .vertical
+    stackView.spacing = 16
+    stackView.addArrangedSubview(titleAndButtonsStackViewContainer)
+    stackView.addArrangedSubview(topLabelStack)
+    stackView.addArrangedSubview(bottomLabelStack)
+    stackView.addArrangedSubview(actionsStackView)
+    return stackView
+  }()
+  
+  private lazy var titleStackView: UIStackView = {
+    let stackView = UIStackView()
+    stackView.axis = .vertical
+    stackView.spacing = 2
+    stackView.alignment = .center
+    stackView.addArrangedSubview(pagesLabel)
+    stackView.addArrangedSubview(addressLabel)
+    stackView.addArrangedSubview(timeframeLabel)
+    return stackView
+  }()
+
+  private lazy var titleAndButtonsStackViewContainer: UIView = {
+    let stackView = UIStackView()
+    stackView.backgroundColor = .nxGrey10
```

Comment:
why do we have nexar color here?
@danageva do we want to allow this?

Reply:
@shaniberacha we have Nexar colors throughout the design.
@danageva if you prefer we can rename each of the components